suggestFragmentのレイアウトの作成

sanadryujiwa opened the pull request:
## 目的
suggestFragmentのレイアウトの作成

## 方針
- Figmaの設計に従ってレイアウトを書く
- 時間の入力は、dateTimePickerを使う

## 実装
- dateTimePickerは、時間入力用のTextViewのOnClickから呼び出す

## テスト
目的, 予算, 時間の３つの項目のそれぞれで、テキスト, 数字, 時間の入力を行えることを確かめる
なお、時間の入力では専用のダイアログが表示される

sanadryujiwa committed:
sign inレイアウトファイル内で命名規則による一部修正

sanadryujiwa committed:
suggestFragmentのレイアウト作成

Ryuki0222 commented on app/src/main/res/layout/fragment_suggest.xml:
```diff
@@ -1,17 +1,203 @@
+            android:paddingStart="@dimen/suggest_text_form_general_padding_horizon"
+            android:paddingEnd="@dimen/suggest_text_form_general_padding_horizon"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toBottomOf="@id/suggestTextPurpose"/>
```
input_typeを追加

Ryuki0222 commented on app/src/main/res/layout/fragment_suggest.xml:
```diff
@@ -1,17 +1,203 @@
+            android:layout_marginStart="@dimen/suggest_text_form_general_margin_start"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintEnd_toStartOf="@id/suggestTextBudgetTilde"
+            app:layout_constraintTop_toBottomOf="@id/suggestTextBudget" />
```
input_typeを追加

Ryuki0222 commented on app/src/main/res/layout/fragment_suggest.xml:
```diff
@@ -1,17 +1,203 @@
+            android:paddingEnd="@dimen/suggest_text_form_general_padding_horizon"
+            app:layout_constraintStart_toEndOf="@id/suggestTextBudgetTilde"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintTop_toBottomOf="@id/suggestTextBudget"/>
```
input typeを追加

Ryuki0222 commented on app/src/main/res/layout/fragment_suggest.xml:
```diff
@@ -1,17 +1,203 @@
+            android:paddingEnd="@dimen/suggest_text_form_general_padding_horizon"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintEnd_toStartOf="@id/suggestTextTimeTilde"
+            app:layout_constraintTop_toBottomOf="@id/suggestTextTime"/>
```
textViewに変更して、timepickerdialogにする

Ryuki0222 commented on app/src/main/res/layout/fragment_suggest.xml:
```diff
@@ -1,17 +1,203 @@
+            android:paddingEnd="@dimen/suggest_text_form_general_padding_horizon"
+            app:layout_constraintStart_toEndOf="@id/suggestTextTimeTilde"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintTop_toBottomOf="@id/suggestTextTime"/>
```
textViewに変更して、timepickerdialogにする

Ryuki0222 commented on app/src/main/res/layout/fragment_suggest.xml:
```diff
@@ -1,17 +1,203 @@
+        android:layout_marginTop="@dimen/suggest_button_margin_top"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintTop_toBottomOf="@id/suggestFrame"/>
```
影つける

Ryuki0222 reviewed (commented):
@sanadryujiwa 
レビューしますた！

sanadryujiwa committed:
TimePickerDialogの実装途中

TimePickerDialogの動作確認ができた段階

sanadryujiwa committed:
suggestFragmentの修正

Commit:
fragment_suggestのフォームの位置を修正

Ryuki0222 commented:
@sanadryujiwa 
https://akira-watson.com/android/datepicker-timepicker.html

Ryuki0222 commented on app/src/main/java/com/example/triplan/ui/suggest/SuggestFragment.kt:
```diff
@@ -1,18 +1,54 @@
 import com.example.triplan.R
+import java.text.SimpleDateFormat
+import java.time.MonthDay
+import java.util.*
```
これ、「*」使うとjava.utilのライブラリ全て読み込むことになるからjava.utilの必要なライブラリだけimport文で記述して

Ryuki0222 commented on app/src/main/java/com/example/triplan/ui/suggest/SuggestFragment.kt:
```diff
@@ -1,18 +1,54 @@
+                cal.set(Calendar.MINUTE, minute)
+                textViewFromTime.text = SimpleDateFormat("HH:mm").format(cal.time)
+            }
+            TimePickerDialog(context, timeSetListener, cal.get(Calendar.HOUR_OF_DAY), cal.get(Calendar.MINUTE), true).show()
```
これdatepickerに変更で！
https://akira-watson.com/android/datepicker-timepicker.html

Ryuki0222 commented on app/src/main/res/layout/fragment_suggest.xml:
```diff
@@ -1,17 +1,216 @@
+            android:paddingEnd="@dimen/suggest_text_form_general_padding_horizon"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintEnd_toStartOf="@id/suggestTextTimeTilde"
+            app:layout_constraintTop_toBottomOf="@id/suggestTextTime"/>
```
デザインに下線が無いので追加

Ryuki0222 commented on app/src/main/res/layout/fragment_suggest.xml:
```diff
@@ -1,17 +1,216 @@
+            android:paddingEnd="@dimen/suggest_text_form_general_padding_horizon"
+            app:layout_constraintStart_toEndOf="@id/suggestTextTimeTilde"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintTop_toBottomOf="@id/suggestTextTime"/>
```
デザインに下線が無いので追加

Ryuki0222 reviewed (commented):
@sanadryujiwa 
レビューしたー

sanadryujiwa committed:
suggestFragmentの修正

- importするパッケージで必要のないものを消した
- timePickerDialogでなく、datePickerDialogを使った
- 日付を入力するTextViewに下線を追加した

sanadryujiwa committed:
Merge branch 'develop' into feature/9KsuOrG3/create_suggest_fragment
